File: 04_Data_Structures_&_Counting/2839_Greedy_Minimization/Answer2839.kt
fun main(){
  val n = readln().toInt()

  for(i in n/5 downTo 0){
    val rest = n - 5 * i
    if(rest % 3 == 0){
      print(i + rest / 3)
      break
    }
    if(i == 0) print(-1)
  }
}

File: 04_Data_Structures_&_Counting/2839_Greedy_Minimization/Answer2839.ts
const n = +require('fs').readFileSync(0)

for(let i = n/5|0; i >=0; i--){
  const rest = n - 5 * i
  if(rest % 3 === 0){
    console.log(i + rest / 3)
    break
  }
  if(i === 0) console.log(-1)
}